File: examples/python/minimal-order-service/adapters/outbound/fake_payment_gateway.py
"""
Outbound adapter: Fake implementation of PaymentGateway.
Always succeeds, used for testing.
"""
import uuid
from application.ports.outbound.payment_gateway import PaymentGateway, PaymentResult
from domain.money import Money


class FakePaymentGateway(PaymentGateway):
    """Fake implementation of PaymentGateway for testing."""
    
    def process_payment(self, amount: Money, order_id: str) -> PaymentResult:
        """
        Process a payment (always succeeds).
        
        Args:
            amount: Amount to charge
            order_id: Order ID
        
        Returns:
            Successful payment result
        """
        transaction_id = f"txn-{uuid.uuid4()}"
        return PaymentResult.success_result(transaction_id)

File: examples/csharp/minimal-order-service/Adapters/Outbound/FakePaymentGateway.cs
using OrderService.Application.Ports.Outbound;
using OrderService.Domain;

namespace OrderService.Adapters.Outbound;

/// <summary>
/// Outbound adapter: Fake implementation of PaymentGateway.
/// Always succeeds, used for testing.
/// </summary>
public class FakePaymentGateway : IPaymentGateway
{
    public PaymentResult ProcessPayment(Money amount, string orderId)
    {
        // Simulate payment processing
        var transactionId = $"txn-{Guid.NewGuid()}";
        return PaymentResult.SuccessResult(transactionId);
    }
}
